boot/bootstate20.go

```diff
@@ -441,21 +462,28 @@ func (bs20 *bootState20Base) markSuccessful(update bootStateUpdate) (bootStateUp
 	return u20, nil
 }
 
-func (bs20 *bootState20Base) setNext(next snap.PlaceInfo) (rbi RebootInfo, u bootStateUpdate, err error) {
-	u20, nextStatus, err := genericSetNext(bs20, next)
+func (bs20 *bootState20Base) setNext(isUndo bool, next snap.PlaceInfo) (rbi RebootInfo, u bootStateUpdate, err error) {
+	u20, rebootRequired, err := genericSetNext(bs20, next)
 	if err != nil {
 		return RebootInfo{RebootRequired: false}, nil, err
 	}
 
-	// if we are setting a snap as a try snap, then we need to reboot
-	rbi.RebootRequired = false
-	if nextStatus == TryStatus {
-		// only update the try base if we are actually in try status
-		u20.writeModeenv.TryBase = next.Filename()
-		// a 'try' base is handled by snap-bootstrap, hence we are not
-		// interested in the bootloader's opinion (no need for
-		// rbi.RebootBootloader, so it is not filled).
-		rbi.RebootRequired = true
+	nextStatus := DefaultStatus
+	rbi.RebootRequired = rebootRequired
+	if rbi.RebootRequired {
+		if isUndo {
+			// we must make sure we boot with the base we revert to
+			u20.writeModeenv.Base = next.Filename()
+			u20.writeModeenv.TryBase = ""
+		} else {
+			// if we need to reboot and we are not undoing, we set the try status
+			// and set appropriately the base we want to try
+			nextStatus = TryStatus
+			u20.writeModeenv.TryBase = next.Filename()
+			// a 'try' base is handled by snap-bootstrap, hence we are not
+			// interested in the bootloader's opinion (no need for
+			// rbi.RebootBootloader, so it is not filled).
```

Comment:
Maybe add a comment also in the `isUndo` branch, because `rbi.RebootBootloader` is not filled there either. Not that I fully understand what this comment says, but it just feels suspicious to have it in one branch of the `if` and not in the other.

Or maybe move it outside of the `if` block, if it applies to both branches.